src/solc_0.8/Game/GameBaseToken.sol

```diff
@@ -65,6 +73,15 @@ contract GameBaseToken is ImmutableERC721, WithMinter, Initializable, IGameToken
         _;
     }
 
+    function onERC721Received(
+        address, /*operator*/
+        address, /*from*/
+        uint256, /*id*/
+        bytes calldata /*data*/
+    ) external pure override returns (bytes4) {
+        return IERC721Receiver.onERC721Received.selector;
```

Comment:
2 things here
- Here we should check that the token sent to the game contract is from our ERC721 asset contract and not another contract.

```require(msgSender() == _asset721)```

- we should check that the tokens transfered to this contract only comes from the calls on the game contract (`createGame`, `updateGame`, etc) and not a manual transfer from the user (those tokens would be stucked then). You can do that by implementing a flag you set when you transfers tokens in this contract  (createGame or updateGame), and you check this flag in `onERC721Received`

```
require(flag[id] == true);
flag[id] = false;
```